app/controllers/users_controller.rb

```diff
@@ -0,0 +1,33 @@
+class UsersController < ApplicationController
+	before_action :authenticate_user!, :validate_current_user
+
+  def show
+  	@user = User.find(params[:id])
+  	@all_projects = Project.all
+  	@all_metrics = ProjectMetrics.metric_names
+  	@preferred_projects = @user.preferred_projects.empty? ? @all_projects : current_user.preferred_projects
+  	@preferred_metrics = @user.preferred_metrics
+  end
+
+  def update_preference
```

Comment:
should we prefer to make this an `update` method to stick to pure RESTful end points?